src/Semantic/IO.hs

```diff
@@ -12,20 +12,22 @@ import           System.Directory (doesDirectoryExist)
 import           System.Directory.Tree (AnchoredDirTree (..))
 import qualified System.Directory.Tree as Tree
 import           System.FilePath
+import qualified System.Path as Path
+import qualified System.Path.PartClass as Path.PartClass
 
 isDirectory :: MonadIO m => FilePath -> m Bool
 isDirectory path = liftIO (doesDirectoryExist path)
 
 -- Recursively find files in a directory.
-findFilesInDir :: MonadIO m => FilePath -> [String] -> [FilePath] -> m [FilePath]
+findFilesInDir :: (Path.PartClass.AbsRel ar, MonadIO m) => Path.Dir ar -> [String] -> [Path.Dir ar] -> m [Path.File ar]
```

Comment:
If you pass in an `AbsDIr`, you will get `AbsFile`s out. If you pass in a `RelDir`, you will get `RelFile`s out. And so on.